serve.go

```diff
@@ -501,6 +506,15 @@ type config struct {
 	err error
 }
 
+func (c *config) setupCommandCallback() {
```

Comment:
I'm fine with this as an experiment but I think that if you introduce a publisher, it should be swappable everywhere the `pubsub` interface is used. 

The idea is to take the existing channel implementation and expose that to multiple consumers, not to create a bunch of random plugins. 

I would suggest that you wrap the inmem pubsub that publishes each event to both the inmem queue and to the callback.